dotnet-interactive.Tests/HttpApiTests.cs

```diff
@@ -198,53 +202,71 @@ public async Task When_subkernel_does_not_exist_then_it_returns_404()
+            var hoverResponse = LspSerializer.JsonSerializer.DeserializeFromString<HoverResponse>(responseJson);
+            hoverResponse.Contents.Kind.Should().Be(Lsp.MarkupKind.Markdown);
+            hoverResponse.Contents.Value.Should().Be(expected);
+        }
+
         [Theory]
         [InlineData(Language.CSharp)]
-        public async Task lsp_textDocument_hover_returns_expected_placeholder(Language language)
+        public async Task lsp_textDocument_hover_returns_no_result_on_malformed_request_object(Language language)
         {
             using var _ = new AssertionScope();
-            var request = JObject.Parse(@"
-{
-    ""textDocument"": {
-        ""uri"": ""path/to/document""
-    },
-    ""position"": {
-        ""line"": 1,
-        ""character"": 2
-    }
-}
-");
-            var response = await GetServer(language).HttpClient.PostJsonAsync("/lsp/textDocument/hover", request);
+            var request = new
+            {
+                SomeField = 1,
+                SomeOtherField = "test",
+            };
+            var response = await GetServer(language).HttpClient.PostObjectAsJsonAsync("/lsp/textDocument/hover", request);
             await response.ShouldSucceed();
             var responseJson = await response.Content.ReadAsStringAsync();
-            var json = JObject.Parse(responseJson);
-            var responseType = json["contents"]["kind"].Value<string>();
-            responseType.Should().Be("markdown");
-            var markdownContent = json["contents"]["value"].Value<string>();
-            markdownContent.Should().Be("textDocument/hover at position (1, 2) with `markdown`");
+            responseJson.Should().Be("null");
         }
 
         [Theory]
         [InlineData(Language.CSharp)]
-        public async Task unimplemented_lsp_methods_return_empty_string(Language language)
+        public async Task lsp_textDocument_hover_returns_no_result_on_empty_request(Language language)
+        {
+            using var _ = new AssertionScope();
+            var response = await GetServer(language).HttpClient.PostObjectAsJsonAsync("/lsp/textDocument/hover", new object());
+            await response.ShouldSucceed();
+            var responseJson = await response.Content.ReadAsStringAsync();
+            responseJson.Should().Be("null");
+        }
+
+        [Theory]
+        [InlineData(Language.CSharp)]
+        [InlineData(Language.FSharp)]
+        [InlineData(Language.PowerShell)]
+        public async Task unimplemented_lsp_methods_return_no_result(Language language)
         {
-            // language kernels that implement LSP handling, but not for the specified method
-            var response = await GetServer(language).HttpClient.PostJsonAsync($"/lsp/not/a/method", new JObject());
+            var response = await GetServer(language).HttpClient.PostObjectAsJsonAsync($"/lsp/not/a/method", new object()); // payload doesn't matter
             await response.ShouldSucceed();
```

Comment:
> await response.ShouldSucceed(); [](start = 11, length = 32)

Maybe a 404?

Reply:
See my other comment about handling HTTP LSP issues in general.